library/src/morpheus-master/js/src/sakai.morpheus.update.presence.js

```diff
@@ -1,4 +1,49 @@
-function updatePresence(){
+var active_tab_listener_attached = false; // initial load
+
+// Set the name of the hidden property and the change event for visibility
+var hidden, visibilityChange;
+if (typeof document.hidden !== "undefined") { // Opera 12.10 and Firefox 18 and later support 
+    hidden = "hidden";
+    visibilityChange = "visibilitychange";
+} else if (typeof document.msHidden !== "undefined") {
+    hidden = "msHidden";
+    visibilityChange = "msvisibilitychange";
+} else if (typeof document.webkitHidden !== "undefined") {
+    hidden = "webkitHidden";
+    visibilityChange = "webkitvisibilitychange";
+}
+
+function updatePresenceTimeout(_ms, _go) {
+    if (window.sakaiLastPresenceTimeOut != null) {
+        clearTimeout(window.sakaiLastPresenceTimeOut);
+        window.sakaiLastPresenceTimeOut = null;
+    }
+    if (_go) {
+        window.sakaiLastPresenceTimeOut = setTimeout('updatePresence()', _ms);
+    }
+}
+
+// Triggers when visibility changes
+function handleVisibilityChange() {
+    if (document.hidden) {
+        console.log("hidden");
+        updatePresenceTimeout(window.sakaiPresenceTimeDelay, false); //stop
+    } else  {
+        console.log("visible");
+        updatePresenceTimeout(window.sakaiPresenceTimeDelay, false); 
+        updatePresence(); //start immediately
+    }
+}
+
+function updatePresence() {
+  if (!active_tab_listener_attached) {
+    document.addEventListener("visibilitychange", handleVisibilityChange, false);
```

Comment:
I think you should be using your variable  "visibilityChange" from earlier. Although perhaps "visibiitychange" just works on all the modern browsers now. Anyway, you should use that variable you created.

Reply:
Yes, can pretty much remove the variable declaration and run it as is.